### DISCUSSION TITLE
Allow for the arbitrary re-ordering of fluid field buttons (and remove "Add" from the buttons)

### DISCUSSION THREAD

**Discussion by murtaugh** [ExpressionEngine/ExpressionEngine#2941] [Feature Requests]
My fluid fields usually contain elements that have some sort of hierarchy. (For example sub-titles and content fields and footnotes.) It's frustrating that often-used buttons can end up at the end of the list, and rarely-used buttons can end up at the beginning, and at the moment there's no way to re-order them (outside of managing to name them alphabetically).

There are a couple possibilities: 

1. Ideally, the fields in the selector would be draggable, which would determine the order of the buttons. This is made difficult by the fact that every field in the system is in that selector. If it were a different kid of selector, a two-column selector, you could have all the fields to be selected from on the left, and then reorder-able selected fields on the right.

2. Re-ordering could be done within the Publish Layout editor.

3. Eliminate "Add" from the fluid field button text. This would allow us to order the button with numbers in our field names (1. Subtitle, 2. Body, etc).

I would argue that "Add" doesn't belong in the button text anyway. It might serve as an educational label the first time someone uses a fluid field, but after that it's unneeded, feels redundant when looking at a large list of "Add... Add... Add..." and gets in the way of usefully creative field-naming.

**Comment by PDXMan1** [comment]
My clients would appreciate these changes also.

**Comment by intoeetive** [comment]
This request hasn't been left without attention: https://github.com/ExpressionEngine/ExpressionEngine/pull/4991
